pkg/controller/persistentvolume/controller_base.go

```diff
@@ -139,14 +139,14 @@ func (ctrl *PersistentVolumeController) initializeCaches(volumeSource, claimSour
 		glog.Errorf("PersistentVolumeController can't initialize caches: %v", err)
 		return
 	}
-	volumeList, ok := volumeListObj.(*api.List)
+	volumeList, ok := volumeListObj.(*api.PersistentVolumeList)
```

Comment:
I have posted an new commit at #28002 that includes unit test changes (work for me).
I really don't mind which patch gets in...